File: 07_app_development/MyProject/libtsparser/src/main/java/com/tosmart/xiexuming/libtsparser/table/ProgramMapTable.java
package com.tosmart.xiexuming.libtsparser.table;

import androidx.annotation.NonNull;

import com.tosmart.xiexuming.libtsparser.descriptor.Descriptor;

import java.util.ArrayList;

/**
 * @author xiexuming
 */
public class ProgramMapTable {
    private ArrayList<Component> components;
    private int pid;
    private ArrayList<Descriptor> programInfo;

    @NonNull
    @Override
    public String toString() {
        return "ProgramMapTable{" +
                "pid=0x" + Integer.toHexString(pid) +
                ", programInfo=" + programInfo +
                "， components=" + components +
                '}' + "\n";
    }

    public ArrayList<Component> getComponents() {
        return components;
    }

    public void setComponents(ArrayList<Component> components) {
        this.components = components;
    }

    public int getPid() {
        return pid;
    }

    public void setPid(int pid) {
        this.pid = pid;
    }

    public ArrayList<Descriptor> getProgramInfo() {
        return programInfo;
    }

    public void setProgramInfo(ArrayList<Descriptor> programInfo) {
        this.programInfo = programInfo;
    }
}

File: newpaserapp/libtosmartparse/src/main/java/com/example/libtosmartparse/table/ProgramMapTable.kt

import com.example.libtosmartparse.table.Component
import com.example.libtosmartparse.descriptor.Descriptor

/**
 * @author xxm
 * 2022/7/17
 **/
class ProgramMapTable {
    private var components: ArrayList<Component>? = null
    private var pid = 0
    private var programInfo: ArrayList<Descriptor>? = null

    override fun toString(): String {
        return """
            ProgramMapTable{pid=0x${Integer.toHexString(pid)}, programInfo=$programInfo，
             components=$components} 
            """.trimIndent()
    }

    fun getComponents(): ArrayList<Component>? {
        return components
    }

    fun setComponents(components: ArrayList<Component>?) {
        this.components = components
    }

    fun getPid(): Int {
        return pid
    }

    fun setPid(pid: Int) {
        this.pid = pid
    }

    fun getProgramInfo(): ArrayList<Descriptor>? {
        return programInfo
    }

    fun setProgramInfo(programInfo: ArrayList<Descriptor>?) {
        this.programInfo = programInfo
    }
}
